homeassistant/components/diagnostics/__init__.py

```diff
@@ -121,17 +123,42 @@ def handle_get(
     )
 
 
-def _get_json_file_response(
+async def _async_get_json_file_response(
+    hass: HomeAssistant,
     data: dict | list,
     filename: str,
+    domain: str,
     d_type: DiagnosticsType,
     d_id: str,
     sub_type: DiagnosticsSubType | None = None,
     sub_id: str | None = None,
 ) -> web.Response:
     """Return JSON file from dictionary."""
+    hass_sys_info = await async_get_system_info(hass)
+    hass_sys_info["run_as_root"] = hass_sys_info["user"] == "root"
+    del hass_sys_info["user"]
+
+    integration: Integration = hass.data[DATA_INTEGRATIONS][domain]
```

Comment:
We should not access hass.data directly but use this:
https://github.com/home-assistant/core/blob/5afbcf4185d9199218fcf404a2667328bc7e5083/homeassistant/loader.py#L564

Reply:
#64752